### PR TITLE
Map sensor rework

### PR DESCRIPTION
**Pull request by 960**

### CODE REVIEW

**Comment by 960** [comment]
Needs enums to be generated

**Comment by rusefi** [comment]
please include generated enums into PR
whatever is needed for PR to be complete - please make this a complete green PR

**Comment by 960** [comment]
done :-)

File: firmware/controllers/algo/rusefi_enums.h [hunk 1]
```diff
@@ -670,42 +670,25 @@ typedef enum {
 } debug_mode_e;
 
 typedef enum {
-	MT_CUSTOM = 0,
-	MT_DENSO183 = 1,
-	/**
-	 * 20 to 250 kPa (2.9 to 36.3 psi) 0.2 to 4.9 V OUTPUT
```
Comment (mck1117) [inline firmware/controllers/algo/rusefi_enums.h]: why reformat and remove comments?

File: firmware/controllers/sensors/map.cpp [hunk 2]
```diff
@@ -42,39 +42,43 @@ static float storedInitialBaroPressure = NAN;
...
 
-static FastInterpolation gm3bar(0.631, 40, 4.914, 304);
+static FastInterpolation mpx4115(0, 10, 5, 118);
 
 static FastInterpolation mpx4250(0, 8, 5, 260);
 
 static FastInterpolation mpx4250A(0.25, 20, 4.875, 250);
 
-static FastInterpolation mpx4100(0.3, 20, 4.9, 105);
+static FastInterpolation mpx5700(0, -31, 5, 746);
```
Comment (mck1117) [inline firmware/controllers/sensors/map.cpp]: can we do something other than this? This is a huge amount of static allocated memory.  Our RAM-poor friends won't be happy :)

File: firmware/controllers/sensors/map.cpp [hunk 3]
```diff
@@ -85,15 +89,24 @@ float decodePressure(float voltage, air_pressure_sensor_config_s * mapConfig DEC
...
-	case MT_SUBY_DENSO:
-	case MT_GM_3_BAR:
-	case MT_TOYOTA_89420_02010:
-	case MT_MPX4100:
+	case               MT_MPX4100:
+	case			   MT_MPX4115:
+	case			   MT_MPX4250:
+	case			   MT_MPX4250A:
+	case			   MT_MPX5700:
+	case			   MT_MPXH6300:
```
Comment (mck1117) [inline firmware/controllers/sensors/map.cpp]: formatting

File: firmware/controllers/sensors/map.cpp [hunk 4]
```diff
@@ -177,24 +189,44 @@ float getBaroPressure(DECLARE_ENGINE_PARAMETER_SIGNATURE) {
...
+				 return &mpx5700;
+			case MT_MPXH6300:
+				 return &mpx6300;
+			case MT_MPX6400:
+				 return &mpx6400;
+			case MT_GM_1_BAR:
+				return &gm1bar;
+			case MT_GM_2_BAR:
+				return &gm2bar;
+			case MT_GM_3_BAR:
```
Comment (mck1117) [inline firmware/controllers/sensors/map.cpp]: formatting

File: firmware/tunerstudio/rusefi.input [hunk 5]
```diff
@@ -1920,12 +1920,14 @@ cmd_set_engine_type_default					= "w\x00\x31\x00\x00"
...
-		field = "MAP ADC input", 						map_sensor_hwChannel
-		field = "MAP type",								map_sensor_type
-		field = "MAP value low point",  		        map_sensor_lowValue
-		field = "MAP voltage low point", 	 		    mapLowValueVoltage
-		field = "MAP value high point",               	map_sensor_highValue
-		field = "MAP voltage high value",               mapHighValueVoltage
+		field = "MAP Sensor Input", 				map_sensor_hwChannel
+		field = "MAP Type",					    	map_sensor_type
+		field = "Linearization Points"
+		field = "MAP Kpa Low",  		            map_sensor_lowValue, {map_sensor_type == 0}
```
Comment (mck1117) [inline firmware/tunerstudio/rusefi.input]: can you share a snip/screenshot of how this looks?

**Comment by rusefi** [comment]
I really like the content of this PR but this should be three separate PRs at least

1) one to change the hard-coded ones
2) another one to expose stuff into TS
3) separate one to change error handling

considering the fact that comments were thrown away and that's not a good decision, this PR will be closed for sure. We can keep it for a couple of days for discussion but this PR would be closed for sure :( Again that's sad considering the positive effort - but if contributor does not play by the rules not much could be done.

One PR should be touching as little as possible.

**Comment by 960** [comment]
I can answer some of the questions before closing.

I reordered all of them to get a nice ordered list in TS.
I changed all the enums, and didnt see any value in adding those comments back in there.
It just looks messy.
 
javaw_wRSjCOX9fQ

javaw_pKwWGAtBiM

I actually dont see any reason to keep this hardcoded at all, as TS has a nice built-in function for it.


**dialog = mapSensorXX, "Calibrate MAP/Baro"
      field = "#MAP Sensor"
field = "MAP type",								map_sensor_type
      field = "MAP Sensor Input", 						map_sensor_hwChannel
	     settingSelector = "Common Map Sensors"
             settingOption = "MPX4115",  map_sensor_lowValue=10,   map_sensor_highValue=118 
             settingOption = "MPX4250",  map_sensor_lowValue=10,   map_sensor_highValue=260
             settingOption = "GM 1-BAR", map_sensor_lowValue=10,   map_sensor_highValue=105
             settingOption = "GM 2-BAR", map_sensor_lowValue=9,  map_sensor_highValue=208
             settingOption = "GM 3-BAR", map_sensor_lowValue=1,  map_sensor_highValue=315 
             settingOption = "MPXH6300", map_sensor_lowValue=1,  map_sensor_highValue=315
             settingOption = "MPX5700", map_sensor_lowValue=-31,  map_sensor_highValue=746
             settingOption = "MPX6400", map_sensor_lowValue=3,  map_sensor_highValue=416
             settingOption = "Denso 079800", map_sensor_lowValue=0,  map_sensor_highValue=173
             settingOption = "VW/Audi/Porsche 250kPa", map_sensor_lowValue=26,  map_sensor_highValue=250
             settingOption = "Bosch 3 Bar TMAP", map_sensor_lowValue=-6,  map_sensor_highValue=323

          field =    "kPa At 0.0 Volts",   map_sensor_lowValue
          field =    "kPa At 5.0 Volts",   map_sensor_highValue**



That also make it easier to add/correct without updating firmware.
I have mentioned this before, but with negative response.
Thats why I tried to make the hardcoded version as nice as possible.

**Comment by 960** [comment]
How it looks with the built in TS function:

javaw_2q7IJrL9Ti


But this includes even more changes :-)

**Comment by rusefi** [comment]
We need hard-coded for better user experience in my opinion

Let's identify smallest isolated change and start creating new PRs?

**Comment by 960** [comment]
If you want hard-coded, we'll use hard-coded..

But you see the reason they all had to be re-ordered/re-done?

I have tested all of them. 
All the settings works fine, and returns the right values.

You know me and git, right? :-)

Could we all agree this is wanted changes, spend the minutes it takes, and cooperate in getting it into place?

I can delete the changed .input if it helps. But the enums (.txt, rusefi_enums and map.cpp) needs to go together for this to work..

The .input/TS are the part I know best of all.

File: firmware/tunerstudio/rusefi.input (hunk 5 above)
Comment (960) [reply firmware/tunerstudio/rusefi.input]: Left some screenshots in another message.

File: firmware/controllers/algo/rusefi_enums.h (hunk 1 above)
Comment (960) [reply firmware/controllers/algo/rusefi_enums.h]: Reordering done for a tidy list in drop-down.

Comments not really useful to anyone in this file..

Would fit better in help-file, TS or documentation

**Comment by mck1117** [comment]
> We need hard-coded for better user experience in my opinion

I don't necessarily agree with that, @rusefi.  What makes you say that? I think it would actually be a better experience if the firmware didn't know about special hardcoded map sensor types, and that information only existed in TS.

> I reordered all of them to get a nice ordered list in TS.

We can't do that.  It breaks backward compatibility.

**Comment by 960** [comment]
> > We need hard-coded for better user experience in my opinion
> 
> I don't necessarily agree with that, @rusefi. What makes you say that? I think it would actually be a better experience if the firmware didn't know about special hardcoded map sensor types, and that information only existed in TS.
> 
> > I reordered all of them to get a nice ordered list in TS.
> 
> We can't do that. It breaks backward compatibility.

Ah, but a new ini version would fix that?
Else it would be ugly :-)

I prefer to use the TS version instead, and have custom hardcoded..

Then it shows you the values in the fields for each sensor..

I dont see any user-expirience benefit from this hard-coded..

**Pull request closed without merging**